js/xt.js

```diff
@@ -0,0 +1,1072 @@
+                'active': true,
+                'contract': true,
+                'linear': linear,
+                'inverse': inverse,
+                'taker': this.safeNumber (market, 'takerFee'),
+                'maker': this.safeNumber (market, 'makerFee'),
+                'contractSize': contractSize,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'feeCurrency': quote,
+                'precision': {
+                    'amount': lot,
+                    'price': step,
+                },
+                'limits': {
+                    'amount': {
+                        'min': minQty,
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': step,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': this.parseNumber (Precise.stringMul (lotString, stepString)),
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#fetchOHLCV
+         * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const options = this.safeValue (this.options, 'timeframes', {});
+        const timeframes = this.safeValue (options, market['type'], {});
+        const timeframeValue = this.safeString (timeframes, timeframe);
+        // spot
+        //
+        // [
+        //    [1664192520, 18821.83, 18822.29, 18815.38, 18817.11, 53.517610, 1007209.60968158]
+        // ]
+        //
+        // swap
+        // [
+        //     {
+        //         "s": "btc_usdt",
+        //         "t": 1664192160000,
+        //         "o": "18859.50",
+        //         "c": "18853.50",
+        //         "h": "18865.00",
+        //         "l": "18853.50",
+        //         "a": "851474",
+        //         "v": "1605870.53130"
+        //     },
+        // ]
+        //
+        if (market['spot']) {
```

Comment:
This method is not quite following our convention of separated request + response, this method should only build+ execute the request (considering the market, etc etc) and then call `parseOHLCVs` that will internally call `parseOHLCV` that should be overridden in each class, please check other implementations for reference.

Reply:
What do I do in the case of the spot fetchOHLCV? The timestamp is in seconds instead of milliseconds

Reply:
@AlgoQ you can simply use `safeTimestamp` that will multiplicate it by 1000